### ISSUE TITLE
Test failing with Julia 1.10.0-beta3

### ISSUE THREAD

**Issue by etpinard** [plotly/Dash.jl#228]
![image](https://github.com/plotly/Dash.jl/assets/6675409/1dcc7444-9c03-4b38-afcd-65a38d02da94)

<details>
<summary>full test log</summary>

```
(Dash) pkg> test
(Dash) pkg> test
     Testing Dash
      Status `/tmp/jl_T60xa2/Project.toml`
⌅ [4c88cf16] Aqua v0.6.7
  [944b1d66] CodecZlib v0.7.2
  [1b08a953] Dash v1.3.0 `~/.julia/dev/Dash`
  [03207cf0] DashBase v1.0.0 `~/.julia/dev/DashBase`
  [1b08a953] DashCoreComponents v2.0.0
  [1b08a953] DashHtmlComponents v2.0.0
  [1b08a953] DashTable v5.0.0
  [864edb3b] DataStructures v0.18.15
  [cd3eb016] HTTP v1.10.0
  [682c06a0] JSON v0.21.4
  [0f8b85d8] JSON3 v1.13.2
  [6ac74813] MD5 v0.2.1
  [ddb6d928] YAML v0.4.9
  [2a0f44e3] Base64 `@stdlib/Base64`
  [44cfe95a] Pkg v1.10.0 `@stdlib/Pkg`
  [6462fe0b] Sockets `@stdlib/Sockets`
  [8dfed614] Test `@stdlib/Test`
  [cf7118a7] UUIDs `@stdlib/UUIDs`
      Status `/tmp/jl_T60xa2/Manifest.toml`
⌅ [4c88cf16] Aqua v0.6.7
  [d1d4a3ce] BitFlags v0.1.7
  [944b1d66] CodecZlib v0.7.2
  [34da2185] Compat v4.10.0
  [f0e56b4a] ConcurrentUtilities v2.2.1
  [1b08a953] Dash v1.3.0 `~/.julia/dev/Dash`
  [03207cf0] DashBase v1.0.0 `~/.julia/dev/DashBase`
  [1b08a953] DashCoreComponents v2.0.0
  [1b08a953] DashHtmlComponents v2.0.0
  [1b08a953] DashTable v5.0.0
  [864edb3b] DataStructures v0.18.15
  [460bff9d] ExceptionUnwrapping v0.1.9
  [cd3eb016] HTTP v1.10.0
  [692b3bcd] JLLWrappers v1.5.0
  [682c06a0] JSON v0.21.4
  [0f8b85d8] JSON3 v1.13.2
  [e6f89c97] LoggingExtras v1.0.3
  [6ac74813] MD5 v0.2.1
  [739be429] MbedTLS v1.1.7
  [4d8831e6] OpenSSL v1.4.1
  [bac558e1] OrderedCollections v1.6.2
  [69de0a69] Parsers v2.7.2
  [aea7be01] PrecompileTools v1.2.0
  [21216c6a] Preferences v1.4.1
  [ae029012] Requires v1.3.0
  [777ac1f9] SimpleBufferStream v1.1.0
  [69024149] StringEncodings v0.3.7
  [856f2bd8] StructTypes v1.10.0
  [3bb67fe8] TranscodingStreams v0.9.13
  [5c2747f8] URIs v1.5.0
  [ddb6d928] YAML v0.4.9
  [94ce4f54] Libiconv_jll v1.17.0+0
  [458c3c95] OpenSSL_jll v3.0.11+0
  [0dad84c5] ArgTools v1.1.1 `@stdlib/ArgTools`
  [56f22d72] Artifacts `@stdlib/Artifacts`
  [2a0f44e3] Base64 `@stdlib/Base64`
  [ade2ca70] Dates `@stdlib/Dates`
  [f43a241f] Downloads v1.6.0 `@stdlib/Downloads`
  [7b1f6079] FileWatching `@stdlib/FileWatching`
  [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils`
  [b27032c2] LibCURL v0.6.4 `@stdlib/LibCURL`
  [76f85450] LibGit2 `@stdlib/LibGit2`
  [8f399da3] Libdl `@stdlib/Libdl`
  [56ddb016] Logging `@stdlib/Logging`
  [d6f4376e] Markdown `@stdlib/Markdown`
  [a63ad114] Mmap `@stdlib/Mmap`
  [ca575930] NetworkOptions v1.2.0 `@stdlib/NetworkOptions`
  [44cfe95a] Pkg v1.10.0 `@stdlib/Pkg`
  [de0858da] Printf `@stdlib/Printf`
  [3fa0cd96] REPL `@stdlib/REPL`
  [9a3f8284] Random `@stdlib/Random`
  [ea8e919c] SHA v0.7.0 `@stdlib/SHA`
  [9e88b42a] Serialization `@stdlib/Serialization`
  [6462fe0b] Sockets `@stdlib/Sockets`
  [fa267f1f] TOML v1.0.3 `@stdlib/TOML`
  [a4e569a6] Tar v1.10.0 `@stdlib/Tar`
  [8dfed614] Test `@stdlib/Test`
  [cf7118a7] UUIDs `@stdlib/UUIDs`
  [4ec0a83e] Unicode `@stdlib/Unicode`
  [deac9b47] LibCURL_jll v8.0.1+1 `@stdlib/LibCURL_jll`
  [e37daf67] LibGit2_jll v1.6.4+0 `@stdlib/LibGit2_jll`
  [29816b5a] LibSSH2_jll v1.11.0+1 `@stdlib/LibSSH2_jll`
  [c8ffd9c3] MbedTLS_jll v2.28.2+1 `@stdlib/MbedTLS_jll`
  [14a3606d] MozillaCACerts_jll v2023.1.10 `@stdlib/MozillaCACerts_jll`
  [83775a58] Zlib_jll v1.2.13+1 `@stdlib/Zlib_jll`
  [8e850ede] nghttp2_jll v1.52.0+1 `@stdlib/nghttp2_jll`
  [3f19e933] p7zip_jll v17.4.0+2 `@stdlib/p7zip_jll`
        Info Packages marked with ⌅ have new versions available but compatibility constraints restrict them from upgrading.
Precompiling project...
  34 dependencies successfully precompiled in 29 seconds. 2 already precompiled.
     Testing Running tests...
Test Summary: | Pass  Total  Time
env           |   17     17  0.3s
Test Summary: | Pass  Total  Time
prefixes      |    5      5  0.0s
Test Summary:  | Pass  Total  Time
lazy + dynamic |   13     13  0.0s
Test Summary:                       | Pass  Total  Time
application resources base registry |   57     57  1.2s
Test Summary:                 | Pass  Total  Time
application resources dynamic |   13     13  0.1s
Test Summary:                    | Pass  Total  Time
application resources components |   11     11  0.1s
Test Summary:                  | Pass  Total  Time
application resources external |   14     14  0.1s
Test Summary:                | Pass  Total  Time
application resources assets |   43     43  0.6s
Test Summary:     | Pass  Total  Time
DevTools creation |   37     37  0.0s
Test Summary: | Pass  Total  Time
DevTools end  |   10     10  0.0s
Test Summary: | Pass  Total  Time
default args  |   16     16  0.0s
Test Summary: | Pass  Total  Time
setted args   |   35     35  0.7s
Test Summary: | Pass  Total  Time
route         |   30     30  0.5s
Test Summary: | Pass  Total  Time
router        |   23     23  0.3s
Test Summary: | Pass  Total  Time
base_handler  |   33     33  0.8s
Test Summary: | Pass  Total  Time
compression   |   13     13  0.4s
Test Summary:    | Pass  Total  Time
resource handler |   14     14  3.1s
Test Summary:        | Pass  Total  Time
external_stylesheets |    4      4  0.1s
Test Summary:    | Pass  Total  Time
external_scripts |    3      3  0.0s
Test Summary: | Pass  Total  Time
url paths     |   10     10  0.0s
Test Summary: | Pass  Total  Time
assets paths  |   13     13  0.2s
Test Summary:                | Pass  Total  Time
suppress_callback_exceptions |    4      4  0.0s
Test Summary: | Pass  Total  Time
meta_tags     |   10     10  0.0s
Test Summary: | Pass  Total  Time
index_string  |    1      1  0.0s
Test Summary:  | Pass  Total  Time
show_undo_redo |    2      2  0.0s
Test Summary:      | Pass  Total  Time
interpolate_string |    1      1  0.1s
Test Summary: | Pass  Total  Time
fingerprint   |    8      8  0.0s
Test Summary:  | Pass  Total  Time
parse_includes |    5      5  0.0s
Test Summary: | Pass  Total  Time
task context  |    7      7  0.2s
Test Summary:          | Pass  Total  Time
multiple tasks context |    5      5  0.2s
Test Summary: | Pass  Total  Time
Components    |   10     10  0.2s
handler: Error During Test at /home/etetreault/.julia/dev/Dash/test/core.jl:33
  Got exception outside of a @test
  MethodError: Cannot `convert` an object of type
    Output{String} to an object of type
    Vector{<:Output}

  Closest candidates are:
    convert(::Type{Vector{<:T}}, ::T) where T<:Dash.Dependency
     @ Dash ~/.julia/dev/Dash/src/app/supporttypes.jl:98
    convert(::Type{T}, ::AbstractArray) where T<:Array
     @ Base array.jl:663
    convert(::Type{T}, ::T) where T
     @ Base Base.jl:84
    ...

  Stacktrace:
    [1] Dash.CallbackDeps(output::Output{String}, input::Input{String}, state::Vector{State})
      @ Dash ~/.julia/dev/Dash/src/app/supporttypes.jl:94
    [2] callback!(func::var"#103#106", app::Dash.DashApp, output::Output{String}, input::Input{String}, state::Vector{State}; prevent_initial_call::Nothing)
      @ Dash ~/.julia/dev/Dash/src/app/callbacks.jl:65
    [3] callback!(func::var"#103#106", app::Dash.DashApp, output::Output{String}, input::Input{String}, state::Vector{State})
      @ Dash ~/.julia/dev/Dash/src/app/callbacks.jl:58
    [4] macro expansion
      @ ~/.julia/dev/Dash/test/core.jl:40 [inlined]
    [5] macro expansion
      @ ~/.julia/juliaup/julia-1.10.0-beta3+0.x64.linux.gnu/share/julia/stdlib/v1.10/Test/src/Test.jl:1577 [inlined]
    [6] top-level scope
      @ ~/.julia/dev/Dash/test/core.jl:34
    [7] include(fname::String)
      @ Base.MainInclude ./client.jl:489
    [8] top-level scope
      @ ~/.julia/dev/Dash/test/runtests.jl:9
    [9] include(fname::String)
      @ Base.MainInclude ./client.jl:489
   [10] top-level scope
      @ none:6
   [11] eval
      @ Core ./boot.jl:383 [inlined]
   [12] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:291
   [13] _start()
      @ Base ./client.jl:552
Test Summary: | Error  Total  Time
handler       |     1      1  1.7s
ERROR: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/etetreault/.julia/dev/Dash/test/core.jl:33
in expression starting at /home/etetreault/.julia/dev/Dash/test/runtests.jl:9
ERROR: Package Dash errored during testing

```

</details>

**Comment by etpinard** [comment]
Same story in Julia `1.10.0-rc3+0`

**Comment by etpinard** [comment]
Same story in Julia [v1.10.0 released on December 25, 2023](https://julialang.org/downloads/#current_stable_release).

I'll release a fix in Dash v1.5.0 today.
